### PR TITLE
chore: improve error logs (#10592)

### PR DESCRIPTION
**Pull request by eogns47**
This PR is related to this issue https://github.com/argoproj/argo-cd/issues/10592

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: server/account/account.go [hunk 1]
```diff
@@ -80,22 +80,22 @@ func (s *Server) UpdatePassword(ctx context.Context, q *account.UpdatePasswordRe
...
 	validPasswordRegexp, err := regexp.Compile(passwordPattern)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to compile password regex: %w", err)
 	}
 
 	if !validPasswordRegexp.Match([]byte(q.NewPassword)) {
 		err := fmt.Errorf("New password does not match the following expression: %s.", passwordPattern)
-		return nil, err
+		return nil, fmt.Errorf("failed to compile password regex: %w", err)
```
Comment (crenshaw-dev) [inline server/account/account.go:93]: The error is already wrapped on the line above. How about we just move the error from line 92 to line 93?

**Review by crenshaw-dev — DISMISSED** [review]

File: server/account/account.go (hunk 1 above)
Comment (eogns47) [reply server/account/account.go:93]: @crenshaw-dev Got it, I'll commit again. 🥹

**Comment by crenshaw-dev** [comment]
@eogns47 looks like the failing unit test has `expected` and `actual` flipped on a `Contains` check. Can you take a look?

**Comment by eogns47** [comment]
> @crenshaw-dev okay, i'll check it now ✋

**Comment by eogns47** [comment]
> @eogns47 looks like the failing unit test has `expected` and `actual` flipped on a `Contains` check. Can you take a look?

solved that problem ☺️

**Comment by eogns47** [comment]
@crenshaw-dev Could you please review this PR when you have time? ☺️

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Pull request merged**